libgadget/sfr_eff.c

```diff
@@ -233,8 +233,11 @@ void cooling_and_starformation(void)
 
     walltime_measure("/Misc");
 
-    stars_spawned = stars_converted = 0;
-    sum_sm = sum_mass_stars = localsfr = 0;
+    stars_spawned = ta_malloc("stars_spawned", int, All.NumThreads);
```

Comment:
Add a FIXME comment these arrays will not be manually declared when we switch to OpenMP 4.7

Reply:
which supports per element array reduction

Reply:
There are several places where that happens - it would tidy the code quite a bit. Unfortunately not yet, gcc openmp 4.5 support is still pretty new. Does openmp provide a way to allocate the temporary arrays on the heap?

Reply:
You can alloc with malloc inside a parallel session before entering for loop. Then do a reduction in a critical section after the for loop, but still in the parallel session.

Reply:
I don't think that would be much cleaner than now. We shall catch these cases as they are touched. So that OpenMP is ready, all of the possible conversions are already marked.